components/Banner/examples/BannerStatusIndicatorExample.m

```diff
@@ -0,0 +1,217 @@
+    _appBarViewController = [[MDCAppBarViewController alloc] init];
+    [_appBarViewController applyPrimaryThemeWithScheme:_containerScheme];
+
+    // Behavioral flags.
+    _appBarViewController.inferTopSafeAreaInsetFromViewController = YES;
+    _appBarViewController.headerView.sharedWithManyScrollViews = YES;
+    _appBarViewController.headerView.canOverExtend = NO;
+    _appBarViewController.shouldAdjustHeightBasedOnHeaderStackView = YES;
+    _appBarViewController.safeAreaDelegate = self;
+
+    self.title = @"Banner";
+    [self addChildViewController:_appBarViewController];
+  }
+  return self;
+}
+
+#pragma mark - UIViewController
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  CGSize bannerViewSize = [self.banner sizeThatFits:self.view.bounds.size];
+
+  CGFloat yOrigin = 0.0f;
+  if (@available(iOS 11.0, *)) {
+    UIViewController *topViewController = [self topViewController];
+    yOrigin = topViewController.view.safeAreaInsets.top;
+    self.additionalSafeAreaInsets = UIEdgeInsetsMake(bannerViewSize.height, 0, 0, 0);
+  } else {
+    // Using Banner as a status indicator is only support on iOS 11.0 and above.
+  }
+
+  self.banner.frame = CGRectMake(0.0f, yOrigin, bannerViewSize.width, bannerViewSize.height);
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.appBarViewController.headerView.trackingScrollView = self.tableView;
+  self.appBarViewController.headerView.observesTrackingScrollViewScrollEvents = YES;
+  [self.view addSubview:self.appBarViewController.view];
+  [self.appBarViewController didMoveToParentViewController:self];
+
+  self.navigationItem.rightBarButtonItem =
+      [[UIBarButtonItem alloc] initWithTitle:@"Banner"
+                                       style:UIBarButtonItemStyleDone
+                                      target:self
+                                      action:@selector(showBanner)];
+}
+
+- (UIStatusBarStyle)preferredStatusBarStyle {
+  // Ensure that our status bar is white.
+  return UIStatusBarStyleLightContent;
+}
+
+- (UIViewController *)childViewControllerForStatusBarStyle {
+  return self.appBarViewController;
+}
+
+- (void)dismissBanner {
+  [self.banner removeFromSuperview];
+  self.banner = nil;
+  [self.view setNeedsLayout];
+}
+
+- (void)showBanner {
+  if (self.banner) {
+    return;
+  }
+  self.banner = [[MDCBannerView alloc] init];
+  [self.banner applyThemeWithScheme:_containerScheme];
+  self.banner.textView.text = @"Global Banner Status Indicator";
+  self.banner.textView.textAlignment = NSTextAlignmentCenter;
+  self.banner.layoutMargins = UIEdgeInsetsZero;
+  self.banner.textView.selectable = NO;
+  self.banner.leadingButton.hidden = YES;
+  self.banner.trailingButton.hidden = YES;
+
+  UIViewController *topViewController = [self topViewController];
+  [topViewController.view addSubview:self.banner];
+  [self.view setNeedsLayout];
+
+  UITapGestureRecognizer *tapGestureRecognizer =
```

Comment:
The banner is not made an accessibility button, so users would not know it can be used to dismiss. Consider adding a Button to the banner to dismiss it instead.